src/Stratis.Bitcoin/P2P/PeerConnectorConnect.cs

```diff
@@ -51,20 +51,16 @@ public PeerConnectorConnectNode(NodeSettings nodeSettings, IPeerAddressManager p
         /// <inheritdoc/>
         public override NetworkAddress FindPeerToConnectTo()
         {
-            PeerAddress peer = null;
-
             foreach (var endPoint in this.NodeSettings.ConnectionManager.Connect)
             {
-                peer = this.peerAddressManager.FindPeer(endPoint.MapToIpv6());
-                if (peer == null)
-                    continue;
+                PeerAddress peerAddress = this.peerAddressManager.FindPeer(endPoint);
+                if (peerAddress != null && !this.IsPeerConnected(peerAddress.NetworkAddress.Endpoint))
+                    return peerAddress.NetworkAddress;
 
-                // If the peer is already connected just continue.
-                if (this.IsPeerConnected(endPoint))
-                    continue;
+                continue;
```

Comment:
yeah ok `continue` is redundant then ;)

Reply:
LOL I deserve a fine here ;)

Reply:
You will pay with a 🍺 and then with some more 🍻 after that.